File: Source/DotNET/Backend.Specs/Features/for_FeatureToggles/when_asking_if_feature_is_on/and_feature_does_not_exist.cs
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Machine.Specifications;
using Moq;
using It = Machine.Specifications.It;

namespace Dolittle.Vanir.Backend.Features.for_FeatureToggles.when_asking_if_feature_is_on
{
    public class and_feature_does_not_exist
    {
        static FeatureToggles toggles;
        static bool result;

        Establish context = () =>
        {
            var provider = new Mock<IFeaturesProvider>();
            provider.Setup(_ => _.Provide()).Returns(new Features());
            toggles = new FeatureToggles(provider.Object);
        };

        Because of = () => result = toggles.IsOn("SomeFeature");

        It should_returns_false = () => result.ShouldBeFalse();
    }
}

File: Source/typescript/features/for_FeatureToggles/when_asking_if_feature_is_on/and_feature_does_not_exist.ts
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { IFeaturesProvider } from '../../IFeaturesProvider';
import { Features } from '../../Features';
import { FeatureToggles } from '../../FeatureToggles';
import { BehaviorSubject } from 'rxjs';

describe('when asking if feature is on and feature does not exist', () => {
    const provider: IFeaturesProvider = {
        features: new BehaviorSubject(new Features())
    };

    const toggles = new FeatureToggles(provider);

    const result = toggles.isOn('SomeFeature');

    it('should return false', () => result.should.be.false);
});
